naive_ralloc/src/lib.rs

```diff
@@ -1,85 +1,42 @@
-#![feature(allocator)]
-#![feature(core_intrinsics)]
-#![feature(linkage)]
+#![feature(alloc)]
+#![feature(allocator_api)]
+#![feature(global_allocator)]
 #![no_std]
-#![allocator]
 
+extern crate alloc;
 #[macro_use]
 extern crate sc;
 
-use core::intrinsics;
+use alloc::allocator::{Alloc, AllocErr, Layout};
 
-// fs/mmap.c
-pub unsafe fn brk(brk: usize) -> usize {
-    syscall!(BRK, brk)
-}
+struct Allocator;
+
+unsafe impl<'a> Alloc for &'a Allocator {
+    unsafe fn alloc(&mut self, layout: Layout) -> Result<*mut u8, AllocErr> {
+        let size = layout.size();
+        let align = layout.align();
```

Comment:
This used to be hardcoded to 16 because there was no alignment information. Now that we do have alignment information I'm not sure if this should directly use the requested alignment or use something like `cmp::max(16, layout.align())`. Thoughts, @tbu- ?

Reply:
Should be good like this (without the `max`).